R/MsBackend.R

```diff
@@ -127,6 +127,10 @@
 #' - `$`, `$<-`: access or set/add a single spectrum variable (column) in the
 #'   backend.
 #'
+#' - `[[`, `[[<-`: access or set/add a single spectrum variable (column) in the
+#'   backend. The default implementation uses `$`, thus these methods don't have
+#'   to be implemented for new classes extending `MsBackend`.
```

Comment:
Did you mean `MsBackendDataFrame` here? cause `MsBackend::$` throws an error *is not implemented for class*.

Reply:
No, I meant it this way. I added a default method for `MsBackend`. Calling `[[` on a `MsBackend` will fail because `$` is not implemented/throws an error. But there is e.g. no need to implement `[[,MsBackendDataFrame` because it uses the `[[,MsBackend` which calls `$,MsBackendDataFrame`.

Reply:
@jorainer - I think the following are still missing `[[,Spectra` and `[[<-,Spectra`.

Reply:
Yes, makes sense to have also `[[,Spectra` and `[[<-,Spectra` implemented. I will add that.

Reply:
Just added the missing methods.